protocol/x/clob/e2e/permissioned_keys_test.go

```diff
@@ -0,0 +1,561 @@
+package clob_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	testtx "github.com/dydxprotocol/v4-chain/protocol/testutil/tx"
+	aptypes "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	"github.com/stretchr/testify/require"
+)
+
+type TestBlockWithMsgs struct {
+	Block uint32
+	Msgs  []TestSdkMsg
+}
+
+type TestSdkMsg struct {
+	Msg            sdk.Msg
+	Authenticators []uint64
+	Fees           sdk.Coins
+	Gas            uint64
+	AccountNum     []uint64
+	SeqNum         []uint64
+	Signers        []cryptotypes.PrivKey
+
+	ExpectedRespCode uint32
+	ExpectedLog      string
+}
+
+func TestPlaceOrder_PermissionedKeys_Failures(t *testing.T) {
+	config := []aptypes.SubAuthenticatorInitData{
+		{
+			Type:   "SignatureVerification",
```

Comment:
Nit (no blocking this PR): did we consider using defining an enum type for the authenticator types? Feels like a lot of wasted payload on these strings. Instead the payload could just be an int

Reply:
let me address this separately. created [this](https://linear.app/dydx/issue/CT-1287/use-enums-for-authenticator-types) ticket to keep track